src/NanoProtocol.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include <nanomsg/nn.h>
+#include <string>
+/*
+ * class for encapsulating nanomsg zero copy allocation/deletion
+ */
+class NanoProtocol {
+public:
+   enum class Protocol {TCP,IPC};
```

Comment:
FYI: For Craig.  To the best of my knowledge local tcp will be translated on Linux to ipc. I am not aware of any performance penalites from when I have discussed this with Ben. I could be wrong and we should find out before we decide what direction to take.